### DISCUSSION TITLE
正确的更新步骤？

### DISCUSSION THREAD

**Discussion by tinychest** [zhongyang219/MusicPlayer2#763] [Q&A]
下载最新的压缩包，解压（注意，千万不要运行程序，不然会创建空白的相关配置文件），直接替换到老版本的根目录。

是这样么？我已经踩过上面的坑了，解压运行程序，替换老版本，导致配置，播放列表全丢失了。 - -！

**Comment by lrisora** [comment]
如果你的老程序是放置在一个有写权限的目录的话，默认行为是将配置放在程序目录
所以如果你没有删除老程序所在的文件夹那么配置并不会丢失

直接在压缩包运行的话
由于实际上是压缩程序将压缩包内容解压到临时目录，
之后exe发现自身在没有写权限的目录（临时目录）运行会自动转为使用
AppData/Roaming/MusicPlayer2目录下的配置（没有会创建新的，这可能是你说的空白配置）
注意，这里退出时不应保存修改到压缩包，否则这个压缩包直接解压替换老版本是有问题的
这会创建global_cfg.ini配置文件，并将程序行为修改为默认使用AppData存储配置
也就是存在于程序所在文件夹的旧配置会被忽略
你可以自行编辑global_cfg.ini将true改为false还原这一点（如果老文件夹没有被删除的话）

**Reply by tinychest** [reply]
首先，谢谢回复！不好意思，我提供的问题细节太少了。

- 我的配置是放在程序目录下的。
- 我个人对于压缩包的习惯是，下载完毕 → 解压 → 删除压缩包，这次在把压缩包内容直接全部替换到老版本程序根目录下之前，双击 exe 运行了一次，导致上面配置丢失的情况。

Windows 中对文件替换，将被替换的文件是直接物理删除的，所以配置是无法找回的，在结合 上述回答 以及 [程序文件说明](https://github.com/zhongyang219/MusicPlayer2/wiki/%E7%A8%8B%E5%BA%8F%E6%96%87%E4%BB%B6%E8%AF%B4%E6%98%8E) 证实了我对配置丢失原因的猜想。

播放器非常简洁好用，希望这个问题能帮大家避坑。

**Reply by lrisora** [reply]
我自己也不太相信更新时配置的安全性，
另外配置会被新程序升级到新版，不保证还能向后兼容
还是建议更新前先备份
